### PR TITLE
Web3

### PR DESCRIPTION
**Pull request by nagarajmanjunath**
* Added web3 support
* Remove parity dependency
* Add function to start RPC endpoint in main.rs

### CODE REVIEW

File: run/web3/src/aaa. [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (b00f) [inline run/web3/src/aaa.:1]: What is this file for?

File: run/cli/Cargo.toml [hunk 2]
```diff
@@ -11,3 +11,4 @@ simple_logger = "1.4.0"
 log = "0.4"
 hex-literal = "0.2.1"
 blockchain = { path = "../blockchain" }
+web3 ={path="../web3"}
```
Comment (b00f) [inline run/cli/Cargo.toml:14]: Please don't change cli. create new crate for web3

File: run/cli/src/main.rs [hunk 3]
```diff
@@ -1,7 +1,8 @@
+extern crate blockchain;
 extern crate durian;
 extern crate ethereum_types;
 extern crate simple_logger;
-extern crate blockchain;
+extern crate web3;
```
Comment (b00f) [inline run/cli/src/main.rs:5]: same here. remove any code from here.

File: run/web3/src/main.rs [hunk 4]
```diff
@@ -1,29 +0,0 @@
-extern crate durian;
-extern crate ethereum_types;
```
Comment (b00f) [inline run/web3/src/main.rs:2]: you should keep this file and put your code here instead of cli.

Comment (b00f) [reply run/web3/src/main.rs:2]: follow the comments

File: run/web3/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+pub mod contract_call;
```
Comment (b00f) [inline run/web3/src/lib.rs:1]: web3 is not a lib. So move these codes to main.rs

File: run/web3/src/types/lib.rs [hunk 6]
```diff
@@ -0,0 +1,2 @@
+mod transaction_request;
```
Comment (b00f) [inline run/web3/src/types/lib.rs:1]: this file is redundant. You can use mod.rs.

File: run/web3/src/aaa. (hunk 1 above)
Comment (nagarajmanjunath) [reply run/web3/src/aaa.:1]: removed

File: run/cli/Cargo.toml (hunk 2 above)
Comment (nagarajmanjunath) [reply run/cli/Cargo.toml:14]: i ll create the separate crate

File: run/web3/src/types/lib.rs (hunk 6 above)
Comment (nagarajmanjunath) [reply run/web3/src/types/lib.rs:1]: solved

**Comment by nagarajmanjunath** [comment]
update my code

File: run/web3/src/lib.rs (hunk 5 above)
Comment (nagarajmanjunath) [reply run/web3/src/lib.rs]: removed

**Pull request merged**